File: classificatoria/m-truque-de-magica.cpp
#include <bits/stdc++.h>
#include <map>
using namespace std;

int make_math(int n, string operacao, int operador){
  int result=0;
  if(operacao=="ADICIONAR"){
    result = n + operador;
  } else if(operacao=="SUBTRAIR"){
    result = n - operador;
  } else if(operacao=="MULTIPLICAR"){
    result = n * operador;
  } else if(operacao=="DIVIDIR"){
    result = n / (operador*1.0);
  }
  return result;
}

int main(){
  string line;
  vector<string> words;
  while (getline(cin, line)){
    if(line == ""){
      break;
    }
      istringstream ss(line);
      istream_iterator<string> begin(ss), end;
      vector<string> words(begin, end);
  }

  int n;
  cin >> n;
  int falha = 0;
  for (size_t i = 1; i <= 100; i++) {
    int result = make_math(i, words[0], stoi(words[1]));
  }
  cout << falha;

  return 0;
}

File: classificatoria/m-truque-de-magica.py
def make_math(n, operacao, operador):
    if(operacao=="ADICIONAR"):
        result = n + operador
    elif(operacao=="SUBTRAIR"):
        result = n - operador
    elif(operacao=="MULTIPLICAR"):
        result = n * operador
    elif(operacao=="DIVIDIR"):
        result = n / (operador*1.0)
    return result

n=input()

dic = {}
try:
    while True:
        raw_answer = raw_input()
        answers = raw_answer.split(' ')
        operacao, operador = answers
        dic[operacao] = int(operador)
except ValueError:
    pass

falhou = 0
for i in range(1,100):
    for op in dic:
        i = make_math(i, op, dic[op])
    if(i < 0 or i%1):
        falhou = falhou + 1

print(falhou)
